### PR TITLE
mcp: surface orphan rate in distillery_relations metrics (#635)

### LINKED ISSUE #635: feat(graph): surface isolated-node / orphan rate in distillery_relations metrics
## Gap

`build_relations_graph` (`src/distillery/graph/builders.py:22-36`) adds nodes only as endpoints of `entry_relations` edges, so entries with no relation never enter the graph. On a 51,724-entry instance the relations subgraph is 120 nodes / 296 edges — **99.77% of entries are invisible to every metric**, and `distillery_relations action=metrics` gives no signal that this is the case. `link_prediction` (Adamic-Adar) is orphan-blind by construction too.

Operators can't see graph health: a near-empty graph and a richly-linked one both "succeed".

## Proposed

- `distillery_relations action=metrics` reports `total_entries`, `graph_node_count`, `edge_count`, and **`orphan_rate`** (= 1 − node_count/total_entries).
- Optional `metric=orphans` returning a sample of unlinked entry IDs (feeds a linking / gap-scan pass).

## Acceptance

- Metrics response includes node/edge/total counts + orphan_rate.
- On a feed-heavy instance the high orphan rate is visible without external counting.

## References
- Diagnostic complement to #629 (auto semantic-link on ingest). Field data: 99.77% orphan rate, gominimal ambient instance, 2026-06-22.
- Code: `src/distillery/graph/builders.py:22-36`, `src/distillery/graph/metrics.py`.

### PR DESCRIPTION
**Pull request by norrietaylor**
## Gap

`distillery_relations action=metrics` reported only `node_count`/`edge_count` over the relations subgraph. On a 51,724-entry instance the subgraph is 120 nodes / 296 edges — 99.77% of entries are invisible to every metric, with no signal of it. Operators couldn't see graph health: a near-empty graph and a richly-linked one both "succeed".

## Fix

`action=metrics` now returns `total_entries`, `graph_node_count`, `edge_count`, and `orphan_rate` (`1 - graph_node_count/total_entries`), plus a new `metric=orphans` returning a capped sample of unlinked entry IDs. Additive only — `node_count`/`edge_count` retained, no new MCP tool.

Two correctness fixes applied after review:

- `orphan_rate` clamps to `[0, 1]`. The graph is built from unfiltered `list_relations()`, so an archived-but-still-linked entry is a graph node but is excluded from `total_entries` (non-archived only), making `graph_node_count > total_entries`. Without the clamp the rate went negative (reproduced -1.0).

```python
return max(0.0, min(1.0, 1.0 - graph_node_count / total_entries))
```

- `total_entries` is scoped with the same entry-side filters (`project`/`tags`/`date_from`/`date_to`) that scope the graph in global scope. Previously the denominator ignored them while the graph was filtered, inflating the rate (reproduced 0.833 for a zero-orphan project). The `metric=orphans` sample is scoped with the same filters for consistency.

```python
total_filters: dict[str, Any] = {"status": _NON_ARCHIVED_STATUSES}
if project is not None:
    total_filters["project"] = project
# ... tags / date_from / date_to
total_entries = await store.count_entries(filters=total_filters)
```

## Acceptance

- [x] **Metrics response includes node/edge/total counts + orphan_rate** — `test_metrics_includes_graph_health_fields` asserts `total_entries`, `graph_node_count`, `edge_count`, `orphan_rate` are present; `test_metrics_response_envelope_shape` checks the envelope.
- [x] **On a feed-heavy instance the high orphan rate is visible without external counting** — `test_metrics_orphan_rate_high_when_few_linked` (8 entries, 2 linked -> `orphan_rate` 0.75); `test_metrics_orphans_returns_unlinked_ids` returns the exact unlinked ID set.
- [x] **orphan_rate never negative (review mustFix)** — `test_metrics_orphan_rate_never_negative_with_archived_linked` (archived-but-linked endpoint, `total_entries=1`, `graph_node_count=2`, asserts `orphan_rate == 0.0`); `test_orphan_rate_clamps_when_nodes_exceed_total` unit test on the pure function.
- [x] **orphan_rate not inflated under entry-side filters (review mustFix)** — `test_metrics_orphan_rate_filtered_uses_scoped_denominator` (fully-linked `project=proj-x` + 10 unrelated entries, asserts `total_entries=2`, `orphan_rate == 0.0`).

## Test plan

```
PYTHONPATH=src python -m pytest tests/test_mcp_tools/test_relations_metrics.py tests/graph/test_metrics.py -q
  -> 32 passed
PYTHONPATH=src python -m pytest tests/test_mcp_tools/ tests/graph/ -q
  -> 122 passed
mypy --strict src/distillery
  -> Success: no issues found in 72 source files
ruff check src tests
  -> All checks passed!
```

Closes #635

## Release Notes

* **New Features**
  * Added an orphan detection metric to identify entries missing from the relations graph.
  * Extended `distillery_relations` metrics output with `total_entries`, `graph_node_count`, and `orphan_rate` (`1 - graph_node_count/total_entries`), plus support for `metric="orphans"` to return sampled unlinked entry IDs.
  * Implemented a sampling cap for orphan ID retrieval.

* **Bug Fixes**
  * Improved `orphan_rate` handling to avoid negative or divide-by-zero results via clamping.

* **Tests**
  * Added regression and edge-case coverage for orphan metrics, scoping, clamping, and sampling limits.

### CODE REVIEW

File: src/distillery/mcp/tools/relations.py [hunk 1]
```diff
@@ -687,6 +737,33 @@ async def _handle_metrics(  # noqa: PLR0911, PLR0912
...
+    # entry-side filters, so the count stays over all non-archived entries.
+    total_filters: dict[str, Any] = {"status": _NON_ARCHIVED_STATUSES}
+    if project is not None:
+        total_filters["project"] = project
+    if tags:
+        total_filters["tags"] = tags
+    if date_from is not None:
+        total_filters["date_from"] = date_from
+    if date_to is not None:
+        total_filters["date_to"] = date_to
```
[the remark this answers is not in the document]
Comment (norrietaylor) [reply src/distillery/mcp/tools/relations.py]: Fixed in a706aa1: for scope="ego" the graph ignores the entry-side filters (project/tags/date_*), so I gated the total_filters construction on scope=="global" — total_entries / orphan_rate / metric="orphans" now count the same unfiltered all-non-archived population the ego subgraph is built over. status=non-archived base filter stays unconditional. Added test_metrics_ego_scope_ignores_entry_filters_for_denominator proving the ego denominator ignores a passed project filter and orphan_rate stays consistent with the graph.

File: tests/test_mcp_tools/test_relations_metrics.py [hunk 2]
```diff
@@ -291,6 +291,149 @@ async def test_metrics_cache_hit_flips_on_second_call(store) -> None:  # type: i
...
+        await _store_entry(store, content=f"orphan {i}")
+
+    data = _parse(
+        await _handle_relations(
+            store, {"action": "metrics", "metric": "orphans", "scope": "global"}
+        )
+    )
+    assert data.get("error") is not True
+    assert len(data["results"]) == 3
+
```
[the remark this answers is not in the document]
Comment (norrietaylor) [reply tests/test_mcp_tools/test_relations_metrics.py:473]: Holding on this one. The module is already marked at file scope via `pytestmark = pytest.mark.unit` (line 18), and no test in this file carries a per-test marker. Adding @pytest.mark.integration to the 6 new tests would double-mark them (module unit + function integration), so they'd be selected by both `pytest -m unit` and `pytest -m integration` — breaking marker-based selection — and would be inconsistent with every other test in the file. These are fast in-memory-store unit tests (no external services), so unit is the correct marker.

**Comment by norrietaylor** [comment]
@coderabbitai review

**Comment by norrietaylor** [comment]
@coderabbitai review
